core/src/main/scala/org/apache/spark/util/ThreadUtils.scala

```diff
@@ -156,4 +156,30 @@ private[spark] object ThreadUtils {
         result
     }
   }
+
+  /**
+   * Construct a new Java ForkJoinPool with a specified max parallelism and name prefix.
+   */
+  def newForkJoinPool(prefix: String, maxThreadNumber: Int): ForkJoinPool = {
+    /**
+     * Extend ForkJoinWorkerThread so we can instantiate it.
+     */
+    class SparkForkJoinWorkerThread(pool: ForkJoinPool) extends ForkJoinWorkerThread(pool) {
```

Comment:
Comments are great of course -- javadoc won't do anything here. It could be a line comment.

I think you might be able to collapse most of this with anonymous classes:

```
    val factory = new ForkJoinWorkerThreadFactory {
      override def newThread(pool: ForkJoinPool) =
        new ForkJoinWorkerThread(pool) {
          setName(prefix + "-" + super.getName)
        }
    }
```